File: Climbing_Stairs/climbing-stairs.cpp
// https://leetcode.com/problems/climbing-stairs/

class Solution {
public:
    int climbStairs(int n) {
        if(n <= 1) {
            return 1;
        }
        
        int prev_step1=1, prev_step2=1;
        int cur = 0;
        
        for(int i=2; i<=n; i++) {
            cur = prev_step1 + prev_step2;
            prev_step1 = prev_step2;
            prev_step2 = cur;
        }
        
        return cur;
    }
};

File: Climbing Stairs/climbing-stairs.go
// https://leetcode.com/problems/climbing-stairs/

func climbStairs(n int) int {
	if n <= 1 {
		return 1
	}

	prev_step1 := 1
	prev_step2 := 1
	cur := 0

	for i := 2; i <= n; i++ {
		cur = prev_step1 + prev_step2
		prev_step1 = prev_step2
		prev_step2 = cur
	}

	return cur
}